app/command_autocomplete.go

```diff
@@ -206,6 +207,14 @@ func parseInputTextArgument(arg *model.AutocompleteArg, parsed, toBeParsed strin
 	// input with a single word
 	index := strings.Index(in, " ")
 	if index == -1 { // typing of the single word argument is not finished
+		if a.Pattern != "" {
+			re := regexp.MustCompile(a.Pattern)
```

Comment:
To be able to use `MustCompile` we have to check the validity of the pattern in the `AutocompleteData.IsValid` method [here](https://github.com/mattermost/mattermost-server/blob/43e606173b8afb56adc3fe280397c0f8301063f0/model/command_autocomplete.go#L227)

Reply:
Ok